static/app/utils/replays/hooks/useReplayList.tsx

```diff
@@ -0,0 +1,171 @@
+import {useCallback, useEffect, useState} from 'react';
+import * as Sentry from '@sentry/react';
+import type {Location} from 'history';
+import {stringify} from 'query-string';
+
+import type {Organization} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {getLocalToSystem, getUserTimezone, getUtcDateString} from 'sentry/utils/dates';
+import EventView from 'sentry/utils/discover/eventView';
+import {decodeInteger} from 'sentry/utils/queryString';
+import {mapResponseToReplayRecord} from 'sentry/utils/replays/replayDataUtils';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+import {useLocation} from 'sentry/utils/useLocation';
+import type {ReplayListLocationQuery, ReplayListRecord} from 'sentry/views/replays/types';
+
+const DEFAULT_LIMIT = 50;
+export const DEFAULT_SORT = '-startedAt';
+
+export const REPLAY_LIST_FIELDS = [
+  'countErrors',
+  'duration',
+  'finishedAt',
+  'id',
+  'projectId',
+  'startedAt',
+  'urls',
+  'user',
+];
+
+type Options = {
+  eventView: EventView;
+  organization: Organization;
+  defaultLimit?: number;
+};
+
+type State = {
+  fetchError: undefined | RequestError;
+  isFetching: boolean;
+  pageLinks: null | string;
+  replays: undefined | ReplayListRecord[];
+};
+
+type Result = State;
+
+function useReplayList({
+  eventView,
```

Comment:
Thoughts on eventually refactoring this so we do not depend on an EventView? It's meant to be used for discover type queries and it might be easier to understand if we're just parsing the query strings ourselves.

Reply:
I brought in EventView mostly because I had to implement `generateEventView` inside of `static/app/views/performance/transactionSummary/transactionReplays/index.tsx`

It seemed like it was helping, and I didn't want to do more rewriting of that page. But we could rewrite that page to remove the need for `generateEventView` and then skip EventView everywhere for sure.